pandas/core/strings.py

```diff
@@ -1090,10 +1090,9 @@ def _wrap_result_expand(self, result, expand=False):
         else:
             index = self.series.index
             if expand:
-                cons_row = self.series._constructor
                 cons = self.series._constructor_expanddim
-                data = [cons_row(x) for x in result]
-                return cons(data, index=index)
+                data = [x if (x is not np.nan) else [None] for x in result]
```

Comment:
just redefine ``cons_row`` as a function. e.g. something like

```
def cons_row(x):
    if np.isnan(x):
         return [ x ]
    return x
```

you don't need to fill